### ISSUE TITLE
errcheck does not work with function types

### ISSUE THREAD

**Issue by mmatczuk** [kisielk/errcheck#154]
Version: 1787c4bee836470bf45018cfbc783650db3c6501

Given a package with function type and another function that accepts and calls the function when the passed function is called the result is not checked.

Example:

```go
package demo

import "errors"

type ProviderFunc func(id int) (*int, error)

func Test(p ProviderFunc) error {
	c, _ := p(0)

	// possible nil pointer deref
	if *c == 1 {
		return errors.New("one")
	}

	return nil
}
```

reports no errors.

Expected to see an error on  line `c, _ := p(0)`.

**Comment by dominikh** [comment]
You have to use the `-blank` flag if you want assignments to `_` to be treated as unchecked errors.

**Comment by mmatczuk** [comment]
Thanks @dominikh, you are absolutely right!
